File: Data Structures & Algorithms/validate-parentheses/submission-17.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        m = {
            ")":"(", 
            "]":"[", 
            "}":"{" 
            }
        
        for n in s:
            if n in m:
                if len(stack) > 0 and stack[-1] == m[n]: 
                    stack.pop()
                else:
                    return False
            else:
                stack.append(n)

        return len(stack) == 0

File: Data Structures & Algorithms/daily-temperatures/submission-17.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        const temps = []; // 2d array = [[tval, tidx]]
        let res = Array(temperatures.length).fill(0);

        for (let i = 0; i < temperatures.length; i++) {
            let t = temperatures[i];
            while (temps.length && temps[temps.length-1][0] < t) {
                const [tVal, tIdx] = temps.pop();
                res[tIdx] = i - tIdx;
            }
            temps.push([t,i]);
        }
        return res;
    }
}
